src/components/TabsItem/TabsItem.tsx

```diff
@@ -1,55 +1,103 @@
 
 export interface TabsItemProps extends React.HTMLAttributes<HTMLElement> {
+  /**
+   * Добавляет иконку слева.
+   *
+   * - Для `mode="default"` используйте иконки размером 24.
+   * - Для всех остальных `mode` используйте иконки размером 20.
+   */
+  before?: React.ReactNode;
+  /**
+   * Добавляет элемент слева от `after`.
+   *
+   * - `React.ReactElement` – либо [`Badge`](https://vkcom.github.io/VKUI/#/Badge) с параметром `mode="prominent"`.
+   *   либо [`Counter`](https://vkcom.github.io/VKUI/#/Counter) с параметрами `mode="prominent" size="s"`.
+   * - `number` – для показа текстового блока с переданным числом.
+   */
+  status?: React.ReactElement | number;
+  /**
+   * Добавляет иконку справа.
+   *
+   * Например, `<Icon16Dropdown />`
+   */
   after?: React.ReactNode;
   selected?: boolean;
+  disabled?: boolean;
 }
 
 /**
  * @see https://vkcom.github.io/VKUI/#/TabsItem
  */
 export const TabsItem = ({
+  before,
   children,
-  selected = false,
+  status,
   after,
+  selected = false,
   ...restProps
 }: TabsItemProps) => {
   const platform = usePlatform();
-  const mode: TabsProps["mode"] = React.useContext(TabsModeContext);
-
-  let ItemTypography =
-    mode === "buttons" || mode === "segmented" ? Subhead : Headline;
+  const { mode, withGaps }: TabsContextProps =
+    React.useContext(TabsModeContext);
+  let statusComponent = null;
 
-  if (platform === VKCOM) {
-    ItemTypography = Text;
+  if (status) {
+    statusComponent =
+      typeof status === "number" ? (
+        <Subhead
+          Component="span"
+          vkuiClass="TabsItem__status TabsItem__status--count"
+          weight="2"
+        >
+          {status}
+        </Subhead>
+      ) : (
+        <span vkuiClass="TabsItem__status">{status}</span>
+      );
   }
 
   return (
     <Tappable
       {...restProps}
-      // eslint-disable-next-line vkui/no-object-expression-in-arguments
-      vkuiClass={classNames(getClassName("TabsItem", platform), {
-        "TabsItem--selected": selected,
-      })}
-      hasActive={mode === "segmented"}
+      vkuiClass={classNames(
+        "TabsItem",
+        (platform === IOS || platform === VKCOM) && `TabsItem--${platform}`,
+        mode && `TabsItem--${mode}`,
+        selected && "TabsItem--selected",
+        withGaps && "TabsItem--withGaps"
+      )}
+      hoverMode="TabsItem--hover"
       activeMode="TabsItem--active"
       focusVisibleMode={mode === "segmented" ? "outside" : "inside"}
+      hasActive={mode === "segmented"}
     >
-      <ItemTypography Component="span" vkuiClass="TabsItem__in" weight="2">
+      {before && <div vkuiClass="TabsItem__before">{before}</div>}
+      <Headline
+        Component="span"
+        vkuiClass="TabsItem__label"
+        level={mode === "default" ? "1" : "2"}
+        weight="2"
+      >
         {children}
-      </ItemTypography>
-      {hasReactNode(after) && <div vkuiClass="TabsItem__after">{after}</div>}
+      </Headline>
+      {statusComponent}
```

Comment:
```suggestion
      {status &&
        <Subhead
            Component="span"
            vkuiClass="TabsItem__status"
            weight="2"
          >
            {status}
          </Subhead>
        }
```

Внутри каунтера по идее есть свой собственный типографический компонент, который будет перебивать все эти определения.

Reply:
Я бы всё-равно не стал бы в лишний раз оборачивать в `Subhead`, чтобы исключить в дальнейшем проблемы
Сегодня `Counter` и `Badge`, а завтра чего ещё придумают передавать